types/block.go

```diff
@@ -213,7 +213,7 @@ func (b *Block) fillHeader() {
 	if b.LastCommitHash == nil {
 		b.LastCommitHash = b.LastCommit.Hash()
 	}
-	if b.DataHash == nil {
+	if b.DataHash == nil || len(b.DataAvailabilityHeader.hash) == 0 {
```

Comment:
When is this condition actually hit if `b.DataAvailabilityHeader.hash` is not `nil`? Should this check not be better as `b.DataAvailabilityHeader.hash == nil`?

Reply:
The important part was that it is hit when the daheader wasn't computed yet. I can change to `b.DataAvailabilityHeader.hash == nil`. I like that `len(b.DataAvailabilityHeader.hash) == 0` also works if e.g. someone set this to `[]byte{}` instead of nil.

TBH, I'm not sure why this change was introduced in the first place, but I think some places in the code computed the data root on an empty (nil) DAHeader instead of computing it first. This might not be a problem anymore when #312 is merged.